src/UI/Component/Item/AppointmentItem.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Item;
+
+/**
+ * Interface AppointmentItem
+ * @package ILIAS\UI\Component\Panel\Listing
+ */
+interface AppointmentItem extends Item {
+	/**
+	 * Get the starting point of the appointment.
+	 * @return \ilDateTime
+	 */
+	public function getFrom();
+
+	/**
+	 * Get the ending point of the appointment.
+	 * @return \ilDateTime
+	 */
+	public function getEnd();
+
+	/**
+	 * Get the color of the calendar containing the item as color code (hex).
+	 * @return string
```

Comment:
I think we need both. On the calendar workshop we talked about the possibility to create custom calendar categories, as far as I remember. This will be the moment we need the possibility to just set a color at the latest.
For the sake of design I would support your color palette idea for cases where we know the possible colors in advance. I'm not happy with the naming as "marker" though. Is this idea of a color palette some general concept we want to introduce for the UI framework? We could need a similar thing for tags.